### DISCUSSION TITLE
is there a way to disable aws-chunked

### DISCUSSION THREAD

**Discussion by WillsonFang** [minio/minio#13003] [Q&A]
From the server perspective instead of the put request, can aws-chunked be disabled?

**Comment by harshavardhana** [comment]
@WillsonFang why do you need to do that?

**Reply by WillsonFang** [reply]
we have Content-Encoding set to gzip, which allows a pre-signed url to grab the file which is compressed and the browser deals with the decompression before the client gets it.

But since minio seems to automatically append aws-chunked on the Content-Encoding its breaking this functionality, it downloads as a compressed file which it tries to open in its native application.
![image](https://user-images.githubusercontent.com/87738055/130020302-5a18d47d-ef4d-4fd6-b07f-aa6120296250.png)
this is an example of a file uploaded with ContentEncoding only set to gzip, it still appends it.

**Reply by harshavardhana** [reply]
> But since minio seems to automatically append aws-chunked on the Content-Encoding its breaking this functionality, it downloads as a compressed file which it tries to open in its native application.
> ![image](https://user-images.githubusercontent.com/87738055/130020302-5a18d47d-ef4d-4fd6-b07f-aa6120296250.png)
> this is an example of a file uploaded with ContentEncoding only set to gzip, it still appends it.

We do not append `aws-chunked` which SDK are you talking about?

**Reply by PhoenixICE** [reply]
AWS S3 SDK v3.7 .net

**Reply by harshavardhana** [reply]
> AWS S3 SDK v3.7 .net

Yeah but that's not our problem, is it? :-) - MinIO server won't add `aws-chunked` content-encoding it is your client SDK that is adding. 

You need to open the issue on AWS S3 SDK

**Reply by PhoenixICE** [reply]
Main problem I guess is this:

Content-Encoding | Set the value to aws-chunked.Amazon S3 supports multiple content encodings. For example:Content-Encoding : aws-chunked,gzip
That is, you can specify your custom content-encoding when using Signature Version 4 streaming API.
Note
Amazon S3 stores the resulting object without the aws-chunked encoding. Therefore, when you retrieve the object, it is not aws-chunked encoded.

https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-streaming.html

**Reply by harshavardhana** [reply]
Yeah probably, there is no way around that unfortunately.

**Reply by PhoenixICE** [reply]
Hmm...I took that to imply the aws-chunked flag is removed when its on S3, would it not be the same as just removing the flag on files uploaded to MinIO?

**Reply by harshavardhana** [reply]
> Hmm...I took that to imply the aws-chunked flag is removed when its on S3, would it not be the same as just removing the flag on files uploaded to MinIO?

Yes we always remove it too @WillsonFang during uploads. How are you uploading?

**Reply by WillsonFang** [reply]
I'm assuming when you ask how, you mean something along the lines of what code I'm using -

via the AWS SDK v3.7

```cs
                AmazonS3Config amazonS3Config = new AmazonS3Config() 
                { 
                    ServiceURL = serviceURL,
                    ForcePathStyle = true,
                };
                using (AmazonS3Client awsClient = new AmazonS3Client(accessKey, secretKey, amazonS3Config))
                {
                    string bucket = this.GetBucketName();
                    string key = this.GetKey();
                    uploadRequest = new PutObjectRequest()
                    {
                        BucketName = bucket,
                        Key = key,
                        ContentType = mimeType,
                        InputStream = memoryStream,
                        Timeout = new TimeSpan(0, 59, 0),
                        ServerSideEncryptionMethod = ServerSideEncryptionMethod.AES256,
                    };
                    uploadRequest.Headers.ContentEncoding = "gzip";
                    uploadRequest.Headers.ContentDisposition = string.Format("attachment; filename=\"{0}{1}\"", this.GetFileName(), this.GetFileExtension());
                    awsClient.PutObject(uploadRequest);
                }

**Reply by harshavardhana** [reply]
and when you do `mc stat --debug alias/bucket/objectname` can you share the output?

**Reply by WillsonFang** [reply]
I'll try get that for you, I'm not part of the infrastructure team, so I don't have access to the server it runs on, unless there is someway to run that from the frontend or via code.

**Reply by harshavardhana** [reply]
You can do `StatObject` and get the Content-Encoding value @WillsonFang

**Reply by WillsonFang** [reply]
Oh if its just the Content-Encoding value does the original screen shot not suffice? The S3 browser would use the same endpoint for getting object stats.
